### PR TITLE
[V3 Audio] Sort local files

### PR DESCRIPTION
**Pull request by laggron42**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes

When using the `[p]local` command, the files were given in a weird order (directly returned from `os.listdir()`) and it was a problem, especially with the playlists whose files are indexed. This PR just add a `sorted()` where the bot must fetch the local tracks.

### CODE REVIEW

**Comment by Nodja** [comment]
I modified my code similarly to this and was gonna make a pull request, but I saw this here.

The only change I would make would be to use `sorted([], key=lambda s: s.casefold())` so that the case doesn't affect file order.

File: redbot/cogs/audio/audio.py [hunk 1]
```diff
@@ -596,11 +596,13 @@ def __init__(self, bot):
...
-            for f in os.listdir(current_folder)
-            if (f.lower().endswith(allowed_files)) and (os.path.isfile(current_folder + f))
-        ]
+        folder_list = sorted(
+            [
+                f
+                for f in os.listdir(current_folder)
+                if (f.lower().endswith(allowed_files)) and (os.path.isfile(current_folder + f))
+            ]
+        )
```
Comment (Tobotimus) [inline redbot/cogs/audio/audio.py:606]: You can use `sorted` directly on a generator expression, so you can remove the square brackets:
```diff
- folder_list = sorted(
-     [
-         f
-         for f in os.listdir(current_folder)
-         if (f.lower().endswith(allowed_files)) and (os.path.isfile(current_folder + f))
-     ]
- )
+ folder_list = sorted(
+     f
+     for f in os.listdir(current_folder)
+     if (f.lower().endswith(allowed_files)) and (os.path.isfile(current_folder + f))
+ )
```

**Review by Tobotimus — CHANGES_REQUESTED** [review]
Just a tiny change suggested :smile:

Also I agree with @Nodja that `casefold` should be used.

File: redbot/cogs/audio/audio.py (hunk 1 above)
Comment (laggron42) [reply redbot/cogs/audio/audio.py:606]: 👍

**Review by Tobotimus — APPROVED** [review]

**Pull request merged**